src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -513,34 +533,65 @@ private[table] class IndexedTableImpl(
     }
   }
 
-  override def optimize(revisionID: RevisionID, options: Map[String, String]): Unit =
-    optimize(revisionID, 1.0, options)
+  override def optimizeUnindexedFiles(
+      unindexedFiles: Seq[String],
+      options: Map[String, String]): Unit = {
+    if (unindexedFiles.isEmpty) return // Nothing to optimize
+    // 1. Load the files from the Staging ID (Unindexed)
+    val files =
+      snapshot.loadIndexFiles(stagingID).filter(f => unindexedFiles.contains(f.path))
+    import files.sparkSession.implicits._
+    // 2. Load the Dataframe, the index status and the schema
+    val filesDF = snapshot.loadDataframeFromIndexFiles(files)
+    val indexStatus = snapshot.loadIndexStatus(stagingID)
+    val schema = metadataManager.loadCurrentSchema(tableID)
+    // 3. In a transaction, update the table with the new data
+    metadataManager.updateWithTransaction(
+      tableID,
+      schema,
+      optimizationOptions(options),
+      append = true) {
+      // Index the data with IndexManager
+      val (data, tableChanges) = indexManager.index(filesDF, indexStatus)
+      // Write the data with DataWriter
+      val newFiles = dataWriter.write(tableID, schema, data, tableChanges)
```

Comment:
We have to make sure that the `newFiles` have `dataChange = false.`
Can you confirm that's the case here?

Reply:
ahhh, nice catch

Reply:
This needs to be fixed. Data writer return by default files with `dataChange = true`

Reply:
done!